docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -887,97 +891,140 @@ def get_docx_variables( text ):
+    plural_prefix = reserved_pluralizers_map[prefix]
+    digit = label_groups[2]
+    index = indexify(digit)
+    # it's just a standalone, like "defendant", or it's a numbered singular
+    # prefix, e.g. user3
+    if label == prefix or label == prefix + digit: 
+        return plural_prefix + index # the vars w/ the index, users[0] or users[2]
+
+    suffix = label_groups[3]
+    # Avoid transforming arbitrary suffixes into attributes
+    if not suffix in reserved_suffixes_map:
+        return label  # return it as is
+
+    # Get the mapped suffix attribute if present, else just use the same suffix
+    suffix_as_attribute = reserved_suffixes_map.get(suffix, suffix)
+    return "".join([plural_prefix, index, suffix_as_attribute])
+
+
+def trigger_gather_string(docassemble_var, 
+                          docx_only_suffixes=generator_constants.DOCX_ONLY_SUFFIXES,
+                          reserved_whole_words=generator_constants.RESERVED_WHOLE_WORDS,
+                          reserved_var_plurals=generator_constants.RESERVED_VAR_PLURALS,
+                          reserved_pluralizers_map=generator_constants.RESERVED_PLURALIZERS_MAP,
+                          reserved_suffixes_map=generator_constants.RESERVED_SUFFIXES_MAP):
+    """For a given set of specific cases, transform a
+    PDF field name into a standardized object name
+    that will be the value for the attachment field."""
+    GATHER_CALL = '.gather()'
+    if docassemble_var in reserved_whole_words:
+        return docassemble_var
+
+    if docassemble_var in reserved_var_plurals:
+        return docassemble_var + GATHER_CALL
 
-  # Break up label into its parts
-  label_groups = get_reserved_label_parts(reserved_prefixes, label)
+    # Everything before the first period and everything from the first period to the end
+    var_parts = re.findall(r'([^.]*)(\..*)*', docassemble_var)
 
-  # If no matches to automateable labels were found,
-  # just use the label as it is
-  if (label_groups is None or label_groups[1] == ''):
-    return label  # return without 
-    
-  # With reserved words, we're always using an index
-  # of the plural version of the prefix of the label
-  prefix = label_groups[1]
-  # Turn any singluars into plurals, e.g. 'user' into 'users'
-  var_start = pluralize_base(reserved_pluralizers_map, prefix)
-  if prefix == label: # it's just a standalone, like "defendant"
-    return var_start # Return the pluralized standalone variable
-
-  digit = label_groups[2]
-  index = indexify(digit)
-
-  # If it's a numbered singluar reserved prefix, e.g. user3
-  if label == prefix + digit:
-    # Return the plural plus the index, e.g. users[2]
-    return var_start + index
+    # test for existence (empty strings result in a tuple)
+    if not var_parts[0]:
+        return False
+    # The prefix, ensuring no key or index
+    prefix = re.sub(r'\[.+\]', '', var_parts[0][0])
+    has_reserved_prefix = prefix in reserved_pluralizers_map.values()
```

Comment:
Apologies for getting so fine-grained - am I correct that this is trying to express 'has reserved plural' as opposed to just a reserved prefix?

Reply:
Please get this fine grained! It is expressing that it's both? I think prefix here is slightly different than [RESERVED_PREFIXES](https://github.com/SuffolkLITLab/docassemble-assemblylinewizard/blob/6e2abfd54e58416c702e9870f27cc3a0c2109ab2/docassemble/assemblylinewizard/generator_constants.py#L46): prefix here is just everything before the `[123]` and attributes, whereas prefix for `RESERVED_PREFIXES` is specifically non-plural prefixes. Renaming suggestions are welcome, this same variable name is used down in `is_reserved_docx_label`.

Reply:
I think part of what I was about to say here will be better addressed in the discussion of expected return values that I just added. Again, sorry about leaving that off the first time.